### ISSUE TITLE
Can use blobfuse2 to mount Azure storage accout as root user, but can't when login as a normal OS user

### ISSUE THREAD

**Issue by huanghaiqing1** [Azure/azure-storage-fuse#1051]

**Comment by huanghaiqing1** [comment]
Which version of blobfuse was used? blobfuse2 with version 2.0.1
Which OS distribution and version are you using? I use opensuse 15.4 and centos 7.9 with latest patches both testing environment and met same issue
If relevant, please share your mount command. blobfuse2 mount /home/cn1hh006/hhq.container.20230206141422 --config-file=/home/cn1hh006/.Azure_cfg_blobfuse2.20230206141422.yaml
What was the issue encountered? I can use similar command to mount as root user, but when switch to normal user, return with no warning info, but can't mount same blob resource from Azure storage account.
Have you found a mitigation/solution? No
Please share logs if available.  See attachment

**Comment by huanghaiqing1** [comment]
<img width="951" alt="20230206" src="https://user-images.githubusercontent.com/30017444/216897613-f7bd738f-b4d0-4f49-82f4-2e5ce4b33c4e.PNG">

**Comment by huanghaiqing1** [comment]
[blobfuse2.log](https://github.com/Azure/azure-storage-fuse/files/10660259/blobfuse2.log)

**Comment by huanghaiqing1** [comment]
Here I give you my testing resule in my opensuse environment as below shows:

Feb  6 15:02:09 autoyast blobfuse2[33779]: 2023/02/06 15:02:09 ==> REQUEST/RESPONSE (Try=1/10.695183708s[SLOW >3s], OpTime=10.695408507s) -- RESPONSE SUCCESSFULLY RECEIVED#012   GET https://bmwtestdata.blob.core.chinacloudapi.cn/hhq?comp=list&delimiter=%2F&maxresults=2&restype=container&timeout=3601#012   Authorization: REDACTED#012   User-Agent: [Azure-Storage-Fuse/2.0.0-preview.1 (openSUSE Leap 15.4) Azure-Storage/0.14 (go1.16.2; linux)]#012   X-Ms-Client-Request-Id: [9f35b845-3f8b-4212-54d5-7280959a63fc]#012   X-Ms-Version: [2020-08-04]#012   x-ms-date: [Mon, 06 Feb 2023 07:01:59 GMT]#012   --------------------------------------------------------------------------------#012   RESPONSE Status: 200 OK#012   Content-Type: [application/xml]#012   Date: [Mon, 06 Feb 2023 07:02:08 GMT]#012   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]#012   X-Ms-Client-Request-Id: [9f35b845-3f8b-4212-54d5-7280959a63fc]#012   X-Ms-Request-Id: [8d7091cb-301e-0054-13f8-39ab3d000000]#012   X-Ms-Version: [2020-08-04]
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_CRIT [mount.go (261)]: Starting Blobfuse2 Mount : 2.0.0-preview.1 on (openSUSE Leap 15.4)
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_CRIT [mount.go (262)]: Logging level set to : LOG_DEBUG
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [libfuse.go (192)]: Libfuse::Configure : libfuse
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_INFO [libfuse.go (226)]: Libfuse::Configure : read-only false, allow-other true, default-perm 511, entry-timeout 120, attr-time 120, negative-timeout 120
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [file_cache.go (171)]: FileCache::Configure : file_cache
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_INFO [file_cache.go (234)]: FileCache::Configure : Using default eviction policy
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_INFO [file_cache.go (243)]: FileCache::Configure : create-empty false, cache-timeout 120, tmp-path /tmp/ramdisk.20230206150149/blobfusetmp.20230206150149
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [attr_cache.go (115)]: AttrCache::Configure : attr_cache
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_INFO [attr_cache.go (135)]: AttrCache::Configure : cache-timeout 7200, symlink false, cache-on-list true
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [azstorage.go (79)]: AzStorage::Configure : azstorage
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [config.go (222)]: ParseAndValidateConfig : Parsing config
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_INFO [config.go (289)]: BlockBlob::ParseAndValidateConfig : using the following proxy address from the config file:
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_INFO [config.go (293)]: BlockBlob::ParseAndValidateConfig : sdk logging from the config file: false
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [config.go (369)]: ParseAndReadDynamicConfig : Reparsing config
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_INFO [config.go (357)]: ParseAndValidateConfig : Account: bmwtestdata, Container: hhq, AccountType: BLOCK, Auth: KEY, Prefix: , Endpoint: https://bmwtestdata.blob.core.chinacloudapi.cn/, ListBlock: 0
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_INFO [config.go (361)]: ParseAndValidateConfig : Retry Config: Retry count 3, Max Timeout 3600, BackOff Time 1, Max Delay 3
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [block_blob.go (138)]: BlockBlob::SetupPipeline : Setting up
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [block_blob.go (123)]: BlockBlob::getCredential : Getting credential
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_DEBUG [azauth.go (77)]: azAuth::getAzAuth : Account: bmwtestdata, AccountType: BLOCK, Protocol: https, Endpoint: https://bmwtestdata.blob.core.chinacloudapi.cn/
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [block_blob.go (222)]: BlockBlob::SetPrefixPath : path
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [block_blob.go (173)]: BlockBlob::TestPipeline : Validating
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [azstorage.go (149)]: AzStorage::Start : Starting component azstorage
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [attr_cache.go (97)]: AttrCache::Start : Starting component attr_cache
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [file_cache.go (126)]: Starting component : file_cache
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [lru_policy.go (109)]: lruPolicy::StartPolicy
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [libfuse.go (123)]: Libfuse::Start : Starting component libfuse
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [libfuse_handler.go (98)]: Libfuse::initFuse : Initializing FUSE
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [libfuse_handler.go (100)]: Libfuse::initFuse : Registering fuse callbacks
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [lru_policy.go (267)]: lruPolicy::ClearCache
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [libfuse_handler.go (104)]: Libfuse::initFuse : Populating fuse arguments
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [libfuse_handler.go (128)]: Libfuse::populateFuseArgs
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_DEBUG [libfuse_handler.go (160)]: Libfuse::populateFuseArgs : opts : blobfuse2
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_DEBUG [libfuse_handler.go (160)]: Libfuse::populateFuseArgs : opts : /home/cn1hh006/hhq.container.20230206150149
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_DEBUG [libfuse_handler.go (160)]: Libfuse::populateFuseArgs : opts : -o
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_DEBUG [libfuse_handler.go (160)]: Libfuse::populateFuseArgs : opts : entry_timeout=120,attr_timeout=120,negative_timeout=120,allow_other
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_DEBUG [libfuse_handler.go (160)]: Libfuse::populateFuseArgs : opts : -f
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_DEBUG [libfuse_handler.go (160)]: Libfuse::populateFuseArgs : opts : -ofsname=blobfuse2
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_DEBUG [libfuse_handler.go (160)]: Libfuse::populateFuseArgs : opts : -okernel_cache
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_INFO [libfuse_handler.go (116)]: Libfuse::initFuse : Mounting with fuse3 library
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_ERR [libfuse_handler.go (119)]: Libfuse::initFuse : failed to mount fuse
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [libfuse.go (136)]: Libfuse::Stop : Stopping component libfuse
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [libfuse_handler.go (174)]: Libfuse::destroyFuse : Destroying FUSE
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [file_cache.go (142)]: Stopping component : file_cache
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [lru_policy.go (138)]: lruPolicy::ShutdownPolicy
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [attr_cache.go (107)]: AttrCache::Stop : Stopping component attr_cache
Feb  6 15:02:11 autoyast blobfuse2[33787]: LOG_TRACE [azstorage.go (158)]: AzStorage::Stop : Stopping component azstorage
Feb  6 15:02:34 autoyast blobfuse2[33522]: LOG_TRACE [cache_policy.go (78)]: cachePolicy::getCacheUsage : /tmp/ramdisk.20230206145413/blobfusetmp.20230206145413
Feb  6 15:02:34 autoyast blobfuse2[33522]: LOG_DEBUG [cache_policy.go (130)]: cachePolicy::getUsagePercentage : current cache usage : 0.000000%!(NOVERB)
Feb  6 15:02:34 autoyast blobfuse2[33522]: LOG_TRACE [lru_policy.go (277)]: lruPolicy::Clear-timeout monitor
Feb  6 15:02:34 autoyast blobfuse2[33522]: LOG_TRACE [lru_policy.go (350)]: lruPolicy::updateMarker
Feb  6 15:02:34 autoyast blobfuse2[33522]: LOG_DEBUG [lru_policy.go (373)]: lruPolicy::deleteExpiredNodes : Starts

**Comment by huanghaiqing1** [comment]
<img width="962" alt="20230206-1" src="https://user-images.githubusercontent.com/30017444/216905581-4836e983-30e0-480d-bd3e-8eb1f2970641.PNG">

**Comment by vibhansa-msft** [comment]
As per above logs you are using allow_other and I see in /etc/fuse.conf you already have enabled allow_other option.
Can you make sure the directory you are using for mount and as temp directory has sufficient permissions for the other user, for test you can set it to 777 and try.
Can you configure "sdk-trace: true" under 'azstorage' section in your config file and retry. Share the updated config file and log file with these changes.

**Comment by huanghaiqing1** [comment]
Hello, I found my problem's reason. In my opensuse 15.4 environment, I found there are fuse.conf and fuse3.conf. In fuse.conf, it's user_allow_other is not coment, but fuse3.conf's user_allow_other is comment. After I un-comment both and now I can use blobfuse2 mount as normal user. In my centos 7 environment, although fuse and fuser3 rpms are both installed, but only fuse.conf exist in /etc/. And user_allow_other's initial setting is commented, so I un-comment the parameter and blobfuse2 mount alow works. So I suppose user_allow_other in all related config file should be explicitly enabled.
